rust/core-lib/src/rpc.rs

```diff
@@ -111,253 +340,141 @@ pub enum EditCommand<'a> {
     SelectAll,
     AddSelectionAbove,
     AddSelectionBelow,
-    Scroll { first: i64, last: i64 },
+    Scroll(LineRange),
     GotoLine { line: u64 },
-    RequestLines { first: i64, last: i64 },
+    RequestLines(LineRange),
     Yank,
     Transpose,
-    Click { line: u64, column: u64, flags: u64, click_count: u64 },
-    Drag { line: u64, column: u64, flags: u64 },
+    Click(MouseAction),
+    Drag(MouseAction),
     Gesture { line: u64, column: u64, ty: GestureType},
     Undo,
     Redo,
-    Cut,
-    Copy,
-    Find { chars: Option<&'a str>, case_sensitive: bool },
-    FindNext { wrap_around: bool, allow_same: bool },
-    FindPrevious { wrap_around: bool },
+    FindNext { wrap_around: Option<bool>, allow_same: Option<bool> },
+    FindPrevious { wrap_around: Option<bool> },
     DebugRewrap,
+    /// Prints the style spans present in the active selection.
     DebugPrintSpans,
 }
 
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[serde(tag = "method", content = "params")]
+/// The edit related requests.
+pub enum EditRequest {
+    /// Cuts the active selection, returning their contents,
+    /// or `Null` if the selection was empty.
+    Cut,
+    /// Copies the active selection, returning their contents or
+    /// or `Null` if the selection was empty.
+    Copy,
```

Comment:
Nice! I'm surprised this works. I assumed `Copy` was a reserved keyword, so I went for an ugly `Copy_` in my own code.